packages/flutter_tools/lib/src/intellij/intellij.dart

```diff
@@ -69,17 +69,67 @@ class IntelliJPlugins {
     return _fileSystem.isDirectorySync(packagePath);
   }
 
-  String _readPackageVersion(String packageName) {
-    final String jarPath = packageName.endsWith('.jar')
-        ? _fileSystem.path.join(pluginsPath, packageName)
-        : _fileSystem.path.join(pluginsPath, packageName, 'lib', '$packageName.jar');
-    final File file = _fileSystem.file(jarPath);
-    if (!file.existsSync()) {
-      return null;
+  ArchiveFile _findPluginXml(String packageName) {
+    final List<String> mainJarPathList = <String>[];
+    if (packageName.endsWith('.jar')) {
+      // package exists(checked in _hasPackage
+      mainJarPathList.add(_fileSystem.path.join(pluginsPath, packageName));
+    } else {
+      final String packageLibPath =
+          _fileSystem.path.join(pluginsPath, packageName, 'lib');
+      if (!_fileSystem.isDirectorySync(packageLibPath)) {
+        return null;
+      }
+      // Collect the files with a file suffix of .jar/.zip that contains the plugin.xml file
+      final List<FileSystemEntity> pluginJarPaths = _fileSystem
+          .directory(_fileSystem.path.join(pluginsPath, packageName, 'lib'))
+          .listSync()
+          .where((FileSystemEntity element) {
+            final String name = element.basename;
+            if (element.statSync().type == FileSystemEntityType.file &&
+                (name.endsWith('.jar') || name.endsWith('.zip'))) {
+              return true;
+            } else {
+              return false;
+            }
+          })
+          .toList();
+
+      if (pluginJarPaths.isEmpty) {
+        return null;
+      }
+      // Prefer file with the same suffix as the package name
+      pluginJarPaths.sort((FileSystemEntity a, FileSystemEntity b) {
```

Comment:
if the names don't match at all is it worth including them in the list?

Reply:
Yes, it is.
The reason is that there are several variations of the jar file containing `META-INF/plugin.xml`.
Currently, we need to check at least three files: `flutter-intellij.jar `/ `flutter-intellij-X.Y.Z.jar` / `flutter-idea-X.Y.Z.jar`.